CONTRIBUTING.md

```diff
@@ -1,23 +1,239 @@
+
+Please carefully consider what you want to say and why you want to say it. The issue may be obvious to you however we do not know what you think so structuring your comments and explaning your reasoning will help us understand your concern. The greater the detail the easier it is for us to implement a change.
+
+## Code Quality
+
+### Project Structure
+
+In order to navigate through your project easily there needs to be a structure that compartmentalizes concepts. This means that code is grouped together based on some concept. For example, here is an example structure that we follow for `Sway` files in the `src` directory.
+
+- `data_structures.sw`
+  - Contains your custom data structures such as
+    - structs
+    - enums
+    - Any trait implementations of your custom structures
+- `errors.sw`
+  - Contains enums that are used in `require(..., MyError::SomeError)`
+  - The enums are split into individual errors e.g. `DepositError`, `OwnerError` etc.
+- `events.sw`
+  - Contains structs which are used inside `log()` statements e.g. `log(Deposit { user, amount });`
+- `interface.sw`
+  - Anything that may be exposed to the user e.g. the `abi` interface for your contract(s)
+- `main.sw`
+  - The entry point to your contract that contains the implementation of your `abi`
+- `utils.sw`
+  - Any private functions (helper functions) that your contracts use inside their functions
+  - Since a contract cannot call its own function we need to abstract away the code into some functions so that they can be used inside many functions of your contract
+
+> *Note*
+> `events.sw` and `interface.sw` might be merged in the future into `interface.sw`
+
+### Code Structure
+
+In order to structure your code there is only 1 rule that you must remember and that is that everything must always be ordered alphabetically - as long as there isn't a dependency error or a better name for a default field of an enum. The reason is that it provides a structure that almost everyone will be familiar with. This means that nobody has to learn how you have decided to structure your code and thus where to look for a variable, function, import etc. This makes it really easy to navigate your entire project and in turn navigate all projects without having to memorize the intricacies of each project.
+
+To be explicit
+
+- All dependencies must be declared in alphabetical order just below the declaration of the program type e.g. below `contract;`
+- All imports must be in alphabetical order and any modules within those imports must be in alphabetical order
+- All functions, structs, enums, fields inside structs and enums, parameters of a function, parameters in documentation etc. must be in alphabetical order in all files (this does not apply to variables inside a function)
+
+Example:
+
+```rust
```

Comment:
I think it would be smart to have this in an external file and included in CI, much like how examples are written in the sway book

Reply:
:) Probably, I did write this off-hand without checking if it even compiles

Reply:
Unfortunately, plain markdown doesn't support that kind of include. We could consider making a Sway Applications Book though!

Reply:
This warrants a separate issue. Will be done in #141